src/main/java/org/broadinstitute/hellbender/tools/exome/ExomeReadCounts.java

```diff
@@ -238,13 +244,13 @@ public void onTraversalStart() {
         if (!intervalArgumentCollection.intervalsSpecified()) {
             throw new UserException("you must indicate the set of exon as input intervals");
         }
-        genomeLocFactory = new GenomeLocParser(sequenceDictionary);
         sampleCollection = new SampleCollection(getHeaderForReads());
 
         logger.log(Level.INFO, "Reading exons locations from intervals...");
-        final List<GenomeLoc> locations = IntervalUtils.genomeLocsFromLocatables(genomeLocFactory,
-                intervalArgumentCollection.getIntervals(sequenceDictionary));
-        exonCollection = new IntervalBackedExonCollection(locations);
+
+        exonCollection = new IntervalBackedExonCollection(
+                intervalArgumentCollection.getIntervals(sequenceDictionary),
+                (contig1,contig2) -> Integer.compare(sequenceDictionary.getSequenceIndex(contig1),sequenceDictionary.getSequenceIndex(contig2)));
```

Comment:
Also, does the order of sorting actually matter?  Or is the fact that their is some ordering the important thing?